samplesheets/assayapps/generic/plugins.py

```diff
@@ -0,0 +1,210 @@
+"""Assay app plugin for samplesheets"""
+
+import re
+from django.conf import settings
+
+from altamisa.constants import table_headers as th
+from samplesheets.plugins import SampleSheetAssayPluginPoint
+from samplesheets.rendering import SIMPLE_LINK_TEMPLATE
+from samplesheets.utils import get_top_header
+from samplesheets.views import MISC_FILES_COLL, RESULTS_COLL
+
+
+# Local constants
+APP_NAME = 'samplesheets.assayapps.generic'
+RESULTS_COMMENT = 'SODAR Assay Link Results'
+MISC_FILES_COMMENT = 'SODAR Assay Link MiscFiles'
+DATA_COMMENT_PREFIX = 'SODAR Assay Row Path'
+DATA_LINK_COMMENT = 'SODAR Assay Link Row'
+
+
+class SampleSheetAssayPlugin(SampleSheetAssayPluginPoint):
+    """Plugin for generic data linking in sample sheets"""
+
+    #: Name (used in code and as unique idenfitier)
+    name = 'samplesheets_assay_generic'
+
+    #: Title
+    title = 'Generic Assay Plugin'
+
+    #: App name for dynamic reference to app in e.g. caching
+    app_name = APP_NAME
+
+    #: Identifying assay fields (used to identify plugin by assay)
+    # NOTE: This assay plugin is accessed by the "SODAR Assay Plugin" override
+    assay_fields = []
+
+    #: Description string
+    description = 'Creates data links from comments in ISA investigation file'
+
+    #: Template for assay addition (Assay object as "assay" in context)
+    assay_template = None
+
+    #: Required permission for accessing the plugin
+    # TODO: TBD: Do we need this?
+    permission = None
+
+    #: Toggle displaying of row-based iRODS links in the assay table
+    display_row_links = True
+
+    @staticmethod
+    def _link_from_comment(cell, header, top_header, target_cols, url):
+        """
+        Creates collection links for targeted columns.
+
+        :param cell: Dict (obtained by iterating over a row)
+        :param header: Column header
+        :param top_header: Column top header
+        :param target_cols: List of column names.
+        :param url: Base URL for link target.
+        """
+        # Special case for Material Names
+        if not isinstance(cell['value'], str) or re.search(
+            '.+ <.*>', cell['value']
+        ):
+            return True
+        elif (
+            top_header['value']
+            in th.DATA_FILE_HEADERS + th.MATERIAL_NAME_HEADERS
+        ) and (header['value'] == 'Name'):
+            cell['link'] = f"{url}/{cell['value']}"
+        elif header['value'].lower() in target_cols:
+            cell['value'] = SIMPLE_LINK_TEMPLATE.format(
+                label=cell['value'],
+                url=f"{url}/{cell['value']}",
+            )
+        return True
```

Comment:
There seems to be a bug in the implementation, as we now return `True` every time the function is called. Instead, you should have `return True` inside each of the `if`/`elif` clauses. Also, the clauses can all be changed to `if`, as we never reach the `elif` anyway once we return from inside that condition.